src/validators/profile.validator.ts

```diff
@@ -35,10 +34,17 @@ export class ProfileValidator {
                     reject();
                 }
                 if (res.statusCode >= 400 || body.teamDomain == null) {
-                    logger.error(new FatalError("The provided team or api key is wrong."));
-                    reject();
+                    options.headers.authorization = `AppKey ${profile.apiToken}`;
+                    request.get(url, options, (err, res) => {
+                        if (res.statusCode === 200) {
+                            resolve("AppKey");
+                        } else {
+                            logger.error(new FatalError("The provided team or api key is wrong."));
+                            reject();
+                        }
+                    });
                 } else {
-                    resolve();
+                    resolve("Bearer");
```

Comment:
I also have one improvement instead of using `Bearer` and `AppKey` strings everywhere. You can make a type for the authenticationType that only allows `Bearer` or `AppKey`. You can insert the following snippet in the `profile.interface.ts` file.
```typescript
export interface Profile {
    name: string;
    team: string;
    apiToken: string;
    authenticationType: AuthenticationType
}

export type AuthenticationType = "Bearer" | "AppKey";
// tslint:disable-next-line:variable-name
export const AuthenticationType: { [key: string]: AuthenticationType } = {
    BEARER: "Bearer",
    APPKEY: "AppKey"
}
```
This would make sure that the value of the authenticationType is always within the defined values and you can also use these constants instead of the plain strings. For example here you would use `AuthenticationType.BEARER` instead of `"Bearer"`.